Add ability to check and fix Django settings

vsalvino opened the pull request:
Fixes #5 

* Add CodeRed Cloud Django settings templates.
* Check for wsgi.py. If it cannot be found, check if the `django_project` might be set incorrectly and offer to set it.
* Check for Django settings file. If it doesn't exist, offer to create it from a `settings.py`.
* Check and rewrite Django settings file(s) if they seem to be missing references to important environment variables such as `VIRTUAL_HOST` or `DB_HOST`.

Housekeeping:
* Move env to webapp `__init__` instead of args on utility methods.
* Upgrade pyinstaller
* Fix formatting of spec file (also include new templates dir)

Note: PATCH-ing the `custom_django_project` in coderedapi currently redeploys the webapp. In this case, that is undesired behavior as we are trying to fix this pre-deployment. We should probably change coderedapi before releasing this feature.

The code here seems quite inelegant and/or side-effect heavy. I guess that is the nature of what we are trying to do, with a lot of dependent nested logic. But open to suggestions on practical ways to organize it better (`api.py` file is getting a bit massive)

vsalvino committed:
Add ability to check and fix Django settings

vsalvino committed:
Include templates in package

jlchilders11 commented on cr/api.py:
```diff
@@ -88,65 +91,273 @@ def __init__(self, handle: str, token: str):
+                raise UserCancelError()
+
+        # The following checks are only for Django-based projects.
+        if self.app_type not in [
```
Django app types is used in a few places, we should move that to some kind of global for ease of use.

vsalvino replied:
I refactored it a bit so the apptype checks are in one place.

jlchilders11 commented:
API.py is getting a bit dense and hard to follow. It seems the overwhelming majority of this size comes from django specific logic checks/fixes, so it would make sense logically to move these checks into a `django_checks.py` file.

vsalvino committed:
Reorganize checks

vsalvino commented:
I split out the individual checks into separate functions in utils. Then split high-level checking on the Webapp into separate functions. I'm a bit happier with it, and it should be mostly test-able now.

Also moved enums into `__init__.py` since they are basically constants.

vsalvino committed:
Increase minimum version of setuptools

vsalvino committed:
Set minimum version of pip

vsalvino committed:
install latest pip in pipeline before installing package

vsalvino committed:
Change order of dependency installation

vsalvino committed:
namespace package configuration for templates
